src/machines/lesson-player-machine.ts

```diff
@@ -333,6 +319,37 @@ export const playerMachine = Machine<
         //   },
         // })
       },
+      assignLessonAndViewer: assign((context, event) => {
+        const lessonAndViewerAssignmentEvents = ['LOAD', 'LOADED', 'VIEW']
+
+        if (!lessonAndViewerAssignmentEvents.includes(event.type)) return {}
+
+        const assignment: {viewer?: any; lesson?: any} = {}
+
+        const lessonData = (event as {lesson?: any})?.lesson || context.lesson
+        const viewerData = (event as {viewer?: any})?.viewer || context.viewer
+
+        if (lessonData) {
+          assignment['lesson'] = lessonData
+        }
+        if (viewerData) {
+          assignment['viewer'] = viewerData
+        }
+
+        return assignment
+      }),
```

Comment:
The `lesson` and `viewer` values in some cases were being overridden with `undefined`. This implementation ensures that they are only getting set if there is something to set them to.